ecs-init/volumes/ecs_volume_plugin.go

```diff
@@ -275,18 +274,17 @@ func (a *AmazonECSVolumePlugin) Remove(r *volume.RemoveRequest) error {
 
 	// remove the volume information
 	delete(a.volumes, r.Name)
-	seelog.Infof("Saving state after removing volume %s", r.Name)
-	// remove the state of deleted volume
-	err = a.state.removeVolume(r.Name)
-	if err != nil {
-		seelog.Errorf("Error saving state")
-		return err
-	}
 	// cleanup the volume's host mount path
 	err = a.CleanupMountPath(vol.Path)
 	if err != nil {
 		seelog.Warnf("Cleaning mount path failed for volume %s: %v", r.Name, err)
 	}
+	seelog.Infof("Saving state after removing volume %s", r.Name)
+	// remove the state of deleted volume
+	err = a.state.removeVolume(r.Name)
+	if err != nil {
+		seelog.Errorf("Error saving state after removing volume %s", r.Name)
```

Comment:
i think it's better to log the error?

Reply:
thats right, will log it.

Reply:
what do you mean in this case?

Reply:
earlier the error was not logged. i have changed it.